### DISCUSSION TITLE
Combobox itemToString prop treats item object as string

### DISCUSSION THREAD

**Discussion by yannieyip** [twilio-labs/paste#793] [Q&A] [answered]
I'm working with the Combobox, doing something very similar to [the example here](https://paste.twilio.design/components/combobox/#combobox-using-option-template).

I'm seeing an error on`itemToString={item => (item ? String(item.name) : null)}` although `item` is defined as a json object that contains string type field `name`.
```
TS2339: Property 'name' does not exist on type 'Item'.   Property 'name' does not exist on type 'string'.
```

Could it be a weird behavior because of Typescript? Any help with getting unblocked would be appreciated. Thanks!

**Comment by serifluous** [comment]
Thanks for the question, @yannieyip ! We're a bit short on engineers today, so it may take a little longer than usual to help get an answer to this.

Will check back in here soon 🤞 Let me know if this is very urgently blocking you, though.

**Comment by SiTaggart** [comment, marked as answer]
Hi @yannieyip 

I suspect this is Typescript errors not being particular helpful, but it is doing what it should be doing in terms of type checking. 

`Item` is typed to be either a string or an object. Currently your code only proves to Typescript that `item` exists. At the point at which you try to get `name` from `item`, Typescript still thinks it could either be a string or an object, so it's complaining. The first part of the error is misleading. I suspect if you check that item is not a string before you pull name off of it, you'll clear the error.

Let us know how you get on.

**Comment by yannieyip** [comment]
@SiTaggart Adding the check got rid of the error. Thank you!
